src/content.js

```diff
@@ -70,7 +73,10 @@ async function buildMetaIndex(contentDir, tagsDir, baseUrl) {
 
   const mdFooter = pages
     .filter(page => page._dir.length > 0)
-    .map(page => `[${page._slug}]: ${page._dirUrl}`)
+    .flatMap(page => [
+      `[${page._slug}]: ${page._dirUrl}`,
+      ...page._headers.map(header => `[${page._slug}#${header.id}]: ${page._dirUrl}#${header.id}`)
+    ])
     .join("\n");
```

Comment:
The cross-page anchor links was implemented by adding more named references to the markdown footer, for example:

```md
[I'm a link][some-page] and [I'm a header link][some-page#a-header]

...

[some-page]: /tags/some-page
[some-page#a-header]: /tags/some-page#a-header
```